lib/make_move.rb

```diff
@@ -5,16 +5,27 @@ def initialize(game_state_gateway:)
     @game_state_gateway = game_state_gateway
   end
 
-  def execute(tile, player)
-    raise TileInvalidError, 'Tile out of bounds' if tile <= 0 || tile >= 10
+  def execute(tile)
+    board = @game_state_gateway.retrieve
+    player = board.current_player
+    check_mark_within_bounds(tile, board)
 
-    old_board = @game_state_gateway.retrieve
-    new_board = old_board
-    if new_board.grid[tile - 1].zero?
-      new_board.grid[tile - 1] = player
-      new_board.swap_player(player)
+    if board.grid[tile - 1].zero?
+      board.grid[tile - 1] = player
+      board.swap_player
+    else
+      raise OccupiedError, 'Cannot make the same move twice'
     end
-    @game_state_gateway.save(new_board)
+    @game_state_gateway.save(board)
     true
   end
 end
+
+def check_mark_within_bounds(tile, board)
```

Comment:
This method is defined outside of the class

Reply:
Thank you, will fix it now.